src/test/java/reporting/listener/ExtentListener.java

```diff
@@ -0,0 +1,78 @@
+package reporting.listener;
+
+import com.aventstack.extentreports.ExtentReports;
+import com.aventstack.extentreports.ExtentTest;
+import com.aventstack.extentreports.Status;
+import com.aventstack.extentreports.markuputils.ExtentColor;
+import com.aventstack.extentreports.markuputils.Markup;
+import com.aventstack.extentreports.markuputils.MarkupHelper;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+import reflection.CustomTest;
+import reporting.ExtentManager;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.Date;
+
+public class ExtentListener implements ITestListener {
+  public static ThreadLocal<ExtentTest> testReport = new ThreadLocal<>();
+  static Date date = new Date();
+  static String fileName = "Report_" + date.toString().replace(":", "_").replace(" ", "_") + ".html";
```

Comment:
It's better to use the `replaceAll` method. 
For example: `date.toString().replaceAll("[:\\s]", "_");`

And such kind of logic is better to move to a separate method to boost readability.